README.md

```diff
@@ -0,0 +1,180 @@
+# graphql-node-resource
+
+This library implements the [Node GraphQL type defined by Relay](https://relay.dev/docs/guides/graphql-server-specification/#object-identification). Relay requires that a GraphQL server
+has a way to identify and refetch GraphQL resources. Its meant to be used within a GraphQL API that fetches its types from a REST API.
+
+
+```mermaid
+graph LR;
+    frontend(frontend using relay)-->graphql-api(GraphQL API);
+    graphql-api-->rest-api(REST API);
+    rest-api-->graphql-api;
+    graphql-api-->frontend;
+```
+_example architecture_
+
+
+## What is a Node?
+
+A `Node` is a refetchable GraphQL resource. All `Node`s must have a _globally unique ID_ and be fetchable via this ID. Relay will refetch these resources in the following way:
+
+```
+node(id: $resource_id) {
+    ... on Foo {
+        bar
+        baz
+    }
+}
+```
+
+## The Node Class
+
+The [NodeType](./src/types/NodeType.ts#) class does two things:
+1. it defines the `Node` type required by Relay
+1. it implements a solution to asynchronously resolve the Node by fetching it from a backing REST API
```

Comment:
suggestion: just want to make sure I understanding the semantics clearly, this is most likely just me making noise... 

1. does it define the `Node` or does it provide the concrete implementation for a `Node` in the relay sense?
2. the solution is just an implementation of the relay gql interface that's got some smarts, right? provides implementation for `node`, `nodes` by asynchronously fetching...?